### ISSUE TITLE
Inconsistency when checking minimum statements

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#13251]
### Description

It seems that a minimum statement for a parameter like `min=0.0` is sometimes checked against 0.0 and sometimes against 1e-10.

### Steps to Reproduce

1. Simulate the following modelica model. (The minimum value for the density type is 0.0). I was not able reproduce this issue with a minimum example not using our BRSL library.
```modelica
model MinTest
Modelica.Units.SI.Density simpleDensity = time-1;
Modelica.Units.SI.Density simpleDensityDiff = abs(time-time);
  inner Rexroth_BRSL.BRSL_World bRSL_World  annotation(
    Placement(transformation(origin = {-66, 90}, extent = {{-10, -10}, {10, 10}})));
  Rexroth_BRSL.Generic.Hydraulics.Sources.Tank tank_1 annotation(
    Placement(transformation(origin = {30, -30}, extent = {{-10, -10}, {10, 10}})));
  Rexroth_BRSL.Haegglunds.Internal.Motors.Database.Atom.Atom_10_8 Atom_10_8_1(internalUsage = false, phi(start = 0, fixed = true), n(start = 0, fixed = true)) annotation(
    Placement(transformation(origin = {-20, 6}, extent = {{-14, -10}, {14, 10}})));
equation
  connect(tank_1.port_A, Atom_10_8_1.port_C) annotation(
    Line(points = {{30, -20}, {-24, -20}, {-24, -4}}, color = {0, 130, 174}));
  connect(tank_1.port_A, Atom_10_8_1.port_D) annotation(
    Line(points = {{30, -20}, {-18, -20}, {-18, -4}}, color = {0, 130, 174}));
  connect(tank_1.port_A, Atom_10_8_1.port_A) annotation(
    Line(points = {{30, -20}, {26, -20}, {26, 27}, {-24, 27}, {-24, 16}}, color = {0, 130, 174}));
  annotation(uses(Rexroth_BRSL(version = "3.1.3")));
end MinTest;
```
The required version 3.1.3 of the BRSL will be provided to you by @niklwors.
2. After simulation the following two warnings are written to the log:
![{7E454637-2489-458B-A8EE-0932C290BB47}](https://github.com/user-attachments/assets/d421f5a1-eabb-44d7-9f74-af13168783c5)
The first one is expected since the value of `simpleDensity` starts at -1 which is smaller than zero.
However one can see that the minimum statement checks against 0.0 in one case and against 1e-10 in the other.

`Atom_10_8_1.haegglundsStaticTables.rhoDelta` and `simpleDensityDiff` are both zero during the whole simulation (see [MinTest.csv](https://github.com/user-attachments/files/17828229/MinTest.csv)) but only for `Atom_10_8_1.haegglundsStaticTables.rhoDelta` an assert is thrown which is not correct in our view.

### Expected Behavior
In general there should not be a warning if a value is zero when its minimum value is zero.

I know that checking for equality is numerically not possible but in my eyes it is not understandable why there is an assert thrown for `Atom_10_8_1.haegglundsStaticTables.rhoDelta` but not for `simpleDensityDiff`.

### Version and OS

 - OpenModelica Version: 1.24.0
 - OS: Windows 11, 64 bit
 - MSL 4.0.0 and BRSL3.1.3

Thanks for looking into this.
Aaron Buntrock - Bosch Rexroth

**Comment by phannebohm** [comment]
This looks like a symbolic issue, not numeric, since we explicitly check against 1e-10, not zero.

I'll see why the min attribute is not correct.

**Comment by casella** [comment]
> This looks like a symbolic issue, not numeric, since we explicitly check against 1e-10, not zero.

I'm not sure I get this. Consider this MWE

model M
  Real x(min = 0, start = 0, fixed = true);
equation
  der(x) = 1;
end M;

If we check against 1e-10, this model should be reported as invalid, which it obviously isn't. My take is that we should check against **-1e-10*nominal.x** or something like that. See discussion in #10477
> 
> I'll see why the min attribute is not correct.

**Comment by phannebohm** [comment]
I currently only have version 2.3.0 of the library, so I cannot test this.

@abuntrock could you have a look at the flat model and see what the `min` attribute says for `Atom_10_8_1.haegglundsStaticTables.rhoDelta` and `simpleDensityDiff`?

If `Atom_10_8_1.haegglundsStaticTables.rhoDelta` is part of an alias set, then possibly one of the aliases has `min` attribute `1e-10`. You can see the found aliases with `-d=debugAlias`, but that might not be easy to parse (the new backend has a much nicer log for that).

I can check this once I have the appropriate version of your library.

**Comment by abuntrock** [comment]
@phannebohm I will check the flat model.
@niklwors should have uploaded the new library version on his fileshare that is accessible to OSMC developers.

**Comment by abuntrock** [comment]
I forgot that I was missing an answer here.

The excerpts from the flattened model look like this 
```modelica
Real simpleDensityDiff(quantity = "Density", unit = "kg/m3", displayUnit = "g/cm3", min = 0.0)
protected Real Atom_10_8_1.haegglundsStaticTables.rhoDelta(quantity = "Density", unit = "kg/m3", displayUnit = "g/cm3", min = 0.0) 
protected Real Atom_10_8_1.haegglundsStaticTables.QComp_toB.rhoBase(quantity = "Density", unit = "kg/m3", displayUnit = "g/cm3", min = 1e-10) = Atom_10_8_1.haegglundsStaticTables.rhoDelta
protected Real Atom_10_8_1.haegglundsStaticTables.QComp_toA.rhoBase(quantity = "Density", unit = "kg/m3", displayUnit = "g/cm3", min = 1e-10) = Atom_10_8_1.haegglundsStaticTables.rhoDelta
```
And in the class corresponding to the `QComp_toA` instance, the min modification is set to `1e-10`. 
So I just forgot to look at aliases from `rhoDelta`.
I will close this ticket then, thanks for the hint.

**Comment by casella** [comment]
@abuntrock so I understand there was really no problem with OMC here, right?

**Comment by abuntrock** [comment]
@casella That's correct. I just did not check the minimum statements of other variables which are defined as equal to the one where I thought `1e-10` was wrong. Sorry for the confusion.
